scan/spec/test_command_generator_spec.rb

```diff
@@ -166,6 +198,32 @@
       end
     end
 
+    it "uses a device without version specifier" do
+      options = { project: "./examples/standard/app.xcodeproj", device: "iPhone 6s" }
+      Scan.config = FastlaneCore::Configuration.create(Scan::Options.available_options, options)
+
+      result = Scan::TestCommandGenerator.generate
+      expect(result).to start_with([
+                                     "set -o pipefail &&",
+                                     "env NSUnbufferedIO=YES xcodebuild",
+                                     "-scheme app",
+                                     "-project ./examples/standard/app.xcodeproj",
+                                     # expect the single highest versioned iOS simulator device available with matching name
+                                     "-destination 'platform=iOS Simulator,id=021A465B-A294-4D9E-AD07-6BDC8E186343'",
+                                     "-derivedDataPath '#{Scan.config[:derived_data_path]}'",
+                                     :build,
+                                     :test
+                                   ])
+    end
+
+    it "rejects devices with versions below deployment target" do
+      options = { project: "./examples/standard/app.xcodeproj", device: "iPhone 5 (8.4)" }
+      Scan.config = FastlaneCore::Configuration.create(Scan::Options.available_options, options)
+
+      result = Scan::TestCommandGenerator.generate
+      # FIXME: expect UI error starting "No simulators found that are equal to the version of specifier"
```

Comment:
> because if there is no simulator found, we can't proceed anyway right?

Not for that location, at least. The error could show for one device in a `--devices` array, but the next `device_string` could resolve. In a different scenario, it could select the default.

Is there a `UI.warning` that would be a better fit than `.error`? Does error end the process?